instaclustr/resource_gcp_vpc_peering.go

```diff
@@ -0,0 +1,166 @@
+package instaclustr
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceGCPVpcPeering() *schema.Resource {
+	return &schema.Resource{
+		Create: GCPresourceVpcPeeringCreate,
+		Read:   GCPresourceVpcPeeringRead,
+		Update: resourceVpcPeeringUpdate,
+		Delete: resourceVpcPeeringDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceVpcPeeringStateImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"vpc_peering_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"cdc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"peer_vpc_network_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_subnet": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnets"},
+			},
+
+			"peer_subnets": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnet"},
+			},
+		},
+	}
+}
+
+func GCPresourceVpcPeeringCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Client
+
+	cdcID, err := VpcPeeringCreate(d, meta)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating VPC peering request: %s", err)
+	}
+	var createData CreateGCPVPCPeeringRequest
+	createData, err = GCPcreateVpcPeeringRequest(d)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating GCP VPC peering request: %s", err)
+	}
+
+	var jsonStr []byte
+	jsonStr, err = json.Marshal(createData)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating GCP VPC peering request: %s", err)
+	}
+	var id string
+	id, err = client.CreateVpcPeering(cdcID, jsonStr)
+	if err != nil {
+		return fmt.Errorf("[Error] Error in creating the cluster: %s", err)
+	}
+	d.SetId(id)
+	d.Set("vpc_peering_id", id)
+	d.Set("cdc_id", cdcID)
+	var vpcPeering *GCPVPCPeering
+	vpcPeering, err = client.GCPReadVpcPeering(cdcID, id)
+	if err != nil {
+		return fmt.Errorf("[Error] Error in reading GCP VPC peering connection: %s", err)
```

Comment:
**[Question]** Is this reading a connection or is it reading the VPC configuration?

Reply:
Here the connection is being read.

Reply:
Ok, in that case, the method should be `GCPReadVpcPeeringConnection`, it's fairly ambiguous otherwise. Also out of curiosity, what details are returned?

Reply:
The read vpc peering func for aws which reads the connection is named `ReadVpcPeering` and keeping that in mind to I renamed this func GCPReadVpcPeering